src/redux/Books/booksSlice.js

```diff
@@ -1,17 +1,68 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { v4 as uuidv4 } from 'uuid';
+import axios from 'axios';
 
-// Books Slice
+const BASE_URL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/bQBzRMehcR3M6emvw1S7';
+
+// Async thunk for fetching books from the Bookstore API
+export const fetchBooks = createAsyncThunk('books/fetchBooks', async () => {
+  const response = await axios.get(`${BASE_URL}/books`);
+  return response.data;
+});
+
+// Async thunk for adding a book using the Bookstore API
+export const addBook = createAsyncThunk('books/addBook', async (bookInfo) => {
+  const newBook = {
+    item_id: uuidv4(),
+    author: bookInfo[1],
+    title: bookInfo[0],
+    category: bookInfo[2],
+  };
+  try {
+    const resp = await axios.post(`${BASE_URL}/books`, newBook);
+    return resp.data;
+  } catch (error) {
+    throw new Error(error.response.data.message);
+  }
+});
+
+// Async thunk for deleting a book using the Bookstore API
+// Async thunk for deleting a book using the Bookstore API
+export const deleteBook = createAsyncThunk('books/deleteBook', async (id) => {
+  await axios.delete(`${BASE_URL}/books/${id}`);
+  return id;
+});
+
+// Redux slice for books
 const booksSlice = createSlice({
   name: 'books',
-  initialState: [],
-  reducers: {
```

Comment:
While you did a good job working on the project requirements Your applicatiom is not rendering data from the api and its because of the following error.

Please take a look at the screenshot below:
![Screenshot 2023-06-24 at 2 32 04 AM](https://github.com/kayjeee/ceactbookstore/assets/79506105/004762d6-b037-4afe-8583-4a0e48264273)

It would be nice if ypur fixed this issue

Reply:
done